.github/workflows/client-celo.yml

```diff
@@ -0,0 +1,79 @@
+name: Go / Celo
+
+on:
+  push:
+    branches:
+      - master
+  pull_request:
+    branches:
+      - master
+  workflow_dispatch:
+
+jobs:
+  build-and-test:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+
+      - name: Set up Docker Buildx
+        uses: docker/setup-buildx-action@v1
+
+      - name: Cache Docker layers
+        uses: actions/cache@v2
+        with:
+          path: /tmp/.buildx-celo-cache
+          key: ${{ runner.os }}-buildx-celo-${{ github.sha }}
+          restore-keys: |
+            ${{ runner.os }}-buildx-celo-
+      # TODO: This step was left here intentionally so we can track disk space
+      # usage for a while. We were trying to fight problems with out of disk space
+      # that happened due to the size of data restored from cache. The cache size
+      # was growing linearly with subsequent workflow runs. We want to observe
+      # available disk space for `/`. Fresh execution starts with 21 GB, we expect
+      # to have no less than 15 GB after the cache is restored.
+      - run: sudo df -h
+
+      - name: Build Docker Build Image
+        uses: docker/build-push-action@v2
+        with:
+          target: gobuild
+          tags: go-build-env-celo
+          build-args: |
+            HOST_CHAIN=celo
+          # REVISION=${{ github.sha }} TODO: set revision only when we want to publish the image
+          # VERSION= ? TODO: Configure version, sample: 1.7.6
+          load: true # load image to local registry to use it in next steps
+          cache-from: type=local,src=/tmp/.buildx-celo-cache
+          cache-to: type=local,dest=/tmp/.buildx-celo-cache
+
+      - name: Create test results directory
+        run: mkdir test-results-celo
+
+      - name: Run Go tests
+        run: |
+          docker run \
+            --volume $GITHUB_WORKSPACE/test-results-celo:/mnt/test-results-celo \
+            --workdir /go/src/github.com/keep-network/keep-ecdsa \
+            go-build-env-celo \
+            gotestsum --junitfile /mnt/test-results-celo/unit-tests-celo.xml -- -tags=celo,musl ./...
+
+      - name: Publish unit test results
+        uses: EnricoMi/publish-unit-test-result-action@v1
+        if: always()
+        with:
+          github_token: ${{ secrets.GITHUB_TOKEN }}
+          files:  ./test-results-celo/unit-tests-celo.xml
+          check_name: Celo Client Test Results # name under which test results will be presented in GitHub (optional)
+          comment_on_pr: false # turns off commenting on Pull Requests
```

Comment:
I don't think there is a need to add `-celo` suffix to `test-results` folder and `unit-tests.xml` file. We don't populate the files anywhere outside the workflow, they 'live' within the runner, only for the duration of the job execution.

Reply:
This a leftover from my trials to fix problems with the test reporter and I wasn't aware of https://github.com/EnricoMi/publish-unit-test-result-action/issues/12 yet. So, I tried to separate everything I can to check whether the situation can be improved. Now, I agree we can drop those chain-specific suffixes.

Reply:
Please see https://github.com/keep-network/keep-ecdsa/commit/6266c7a884d8ab3cdd1d68c65c03b4a7df94356f

Reply:
Actually, I think we could clean it up even a little more. We no longer need the step that creates the test results directory. Also `--volume $GITHUB_WORKSPACE/test-results:/mnt/test-results \` and `--junitfile /mnt/test-results/unit-tests.xml` fragments are redundant.
Those lines of code were introduced to pass the test-results from docker image to step publishing the tests results

Reply:
Yeah, I know but I thought we want to keep them as we will return to publishing step results once the bug in the publisher is resolved or we'll switch to another publisher. Anyway, got rid of that for now: https://github.com/keep-network/keep-ecdsa/commit/4287e7f38e5d6d939bf758c2bad56e0268d8a34b